autoload/neomake.vim

```diff
@@ -368,9 +371,9 @@ function! s:AddExprCallback(maker) abort
 
     if list_modified
         if file_mode
-            call setloclist(a:maker.winnr, list, 'r')
+            call setloclist(a:maker.winnr, list, 'r', name)
```

Comment:
Oh, I see. Unfortunately I'm of no help here, since I haven't dug into neomake so far. Hope you figure it out nevertheless.